<a href="https://colab.research.google.com/github/vishal2004-MH/AIML-/blob/main/Vishal_Shinde_P5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Pro 5: To develop classification model and evaluate its performance**


In [ ]:
# Importing required libraries
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score


In [ ]:
# Load Dataset
df= pd.read_csv('ipl_data.csv')
df.head()


Batting Decision       Ground Name Winning Side
0              Bat      Eden Gardens       Team A
1             Bowl  Wankhede Stadium       Team B
2              Bat      Eden Gardens       Team A
3             Bowl  Wankhede Stadium       Team B
4              Bat      Eden Gardens       Team B

In [ ]:
df.tail()

Batting Decision       Ground Name Winning Side
59             Bowl  Wankhede Stadium       Team B
60              Bat      Eden Gardens       Team A
61             Bowl  Wankhede Stadium       Team B
62              Bat      Eden Gardens       Team A
63             Bowl  Wankhede Stadium       Team B

In [ ]:
# Encoding categorical variables
df['Batting Decision'] = df['Batting Decision'].map({'Bat': 0, 'Bowl': 1})
df['Winning Side'] = df['Winning Side'].map({'Team A': 0, 'Team B': 1})
df['Ground Name'] = df['Ground Name'].astype('category').cat.codes


In [ ]:
# Splitting data into training and testing sets
X = df[['Batting Decision', 'Ground Name']]
y = df['Winning Side']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)


In [ ]:
# Creating and training Decision Tree Classifier
clf = DecisionTreeClassifier()
clf.fit(X_train, y_train)


DecisionTreeClassifier()

In [ ]:
# Making predictions
y_pred = clf.predict(X_test)
# Calculating accuracy
accuracy = accuracy_score(y_test, y_pred)
print("Accuracy:", accuracy)


Accuracy: 0.7692307692307693


In [ ]:
# Example prediction
batting_decision = 1 # 0 for Bat, 1 for Bowl
ground_name = 'Eden Gardens' # 0 for Eden Gardens, 1 for Wankhede Stadium
#The following line was a comment before, separating the assignment to 'prediction'. It has been uncommented.
prediction = clf.predict([[0, 0]])
if prediction[0] == 0:
  print("The predicted winning side is Team A")
else:
  print("The predicted winning side is Team B")

The predicted winning side is Team A


/usr/local/lib/python3.11/dist-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but DecisionTreeClassifier was fitted with feature names
  warnings.warn(


In [ ]:
# Evaluate performance using classification report
from sklearn.metrics import classification_report, confusion_matrix
print("Classification Report:")
print(classification_report(y_test, y_pred))


Classification Report:
              precision    recall  f1-score   support

           0       0.80      0.67      0.73         6
           1       0.75      0.86      0.80         7

    accuracy                           0.77        13
   macro avg       0.78      0.76      0.76        13
weighted avg       0.77      0.77      0.77        13



In [ ]:
# Evaluate performance using confusion matrix
print("Confusion Matrix:")
print(confusion_matrix(y_test, y_pred))


Confusion Matrix:
[[4 2]
 [1 6]]


Conclusion:

We successfully developed a classification model using the Decision Tree algorithm and evaluated its performance on a given dataset. The model achieved an accuracy of 0.77 on the testing set. Additionally, the classification report provides insights into the model's performance across different classes, including precision, recall, and F1-score.
